src/core/util/threadpool.cpp

```diff
@@ -43,18 +43,26 @@ ThreadPool::ThreadPool(size_t threads, std::function<void()> onThreadStart,
     }
 }
 
+ThreadPool::~ThreadPool() {
+    std::lock_guard<std::mutex> lock(workers_mutex);
+    for (auto& worker : workers) worker->state = State::Abort;
+    condition.notify_all();
+    workers.clear();  // this will join all threads.
```

Comment:
```suggestion
    std:: unique_lock lock(workers_mutex);
    for (auto& worker : workers) worker->state = State::Abort;
    condition.notify_all();
    auto localCopy = std::move(workers);
    lock.unlock();
    localCopy();  // this will join all threads.
```
This might be a solution.